tensorflow_encrypted/protocol/pond.py

```diff
@@ -52,25 +47,44 @@ def __init__(
         server_0: Player,
         server_1: Player,
         crypto_producer: Player,
-        use_noninteractive_truncation: bool=False
+        use_noninteractive_truncation: bool=False,
+        factory: AbstractFactory=Int100Factory()
     ) -> None:
         self.server_0 = server_0
         self.server_1 = server_1
         self.crypto_producer = crypto_producer
         self.use_noninteractive_truncation = use_noninteractive_truncation
+        self.factory = factory
```

Comment:
could we call it something else, maybe simply `backing_tensor` or `tensor_factory`?